### ISSUE TITLE
Nodemailer SMTP host throws error 

### ISSUE THREAD

**Issue by ShayanAkhun** [DefinitelyTyped/DefinitelyTyped#67197]
SO im doing a project for myself and when creating a transporter the host throws the following error 
**No overload matches this call.
  The last overload gave the following error.
    Argument of type '{ host: string; port: number; service: string | undefined; auth: { user: string | undefined; password: string | undefined; }; }' is not assignable to parameter of type 'TransportOptions | Transport<any>'.
      Object literal may only specify known properties, and 'host' does not exist in type 'TransportOptions | Transport<any>'.ts(2769)
index.d.ts(77, 17): The last overload is declared here.
(property) SMTPConnection.Options.host?: string | undefined**
and my code is

 `require("dotenv").config()
import nodemailer, {Transporter} from 'nodemailer'
import ejs from 'ejs'
import path from 'path'

interface EmailOptions {
    email:string,
    subject:string,
    template: string,
    data :{[key:string]:any}
}

const sendMail = async (options: EmailOptions) : Promise <void> => {
    const transporter : Transporter = nodemailer.createTransport({
        host : process.env.SMTP_HOST as string,
        port: parseInt(process.env.SMTP_PORT || '587') ,
        service: process.env.SMTP_SERVICE ,
        auth : {
            user: process.env.SMTP_EMAIL,
            password: process.env.SMTP_PASSWORD,
        }
    })
    const {email ,subject , template , data} = options

    //get the path to the email template
    const templatePath = path.join(__dirname, '../mails', template);

    //render the email template with ejs
    const html:string = await ejs.renderFile(templatePath,data)

    const mailOptions = {
        from : process.env.SMPT_MAIL,
        to: email,
        subject,
        template
    }
    await transporter.sendMail(mailOptions)
}

export default sendMail
`
![code](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/76954640/2d9e585a-49a6-4e96-989d-d3991333a558)

**Comment by ShayanAkhun** [comment]
So quick upadate... I found a solution for this but it'
![code](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/76954640/427f39d6-0845-4c67-8e97-1029734f153c)
![code](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/76954640/61026ee4-2378-4b47-ad77-0362d3428333)
s not recommended, so i think there are some types missig in tbe types file the path is `node_modules>@types>nodemailer>index.d.ts>and the Function TransportOptions` and you have to add the following types

**Comment by euanmillar** [comment]
I am getting the same issue if I set tls options

**Comment by pranavburnwal** [comment]
This still needs a proper solution with the fix.

**Comment by chrisvire** [comment]
I believe what I wrote above is the proper solution, to use a `SMTPTransport.Options` instance as a parameter to the `nodemailer.createTransport()` method.

**Comment by oxwazz** [comment]
> I believe what I wrote above is the proper solution, to use a `SMTPTransport.Options` instance as a parameter to the `nodemailer.createTransport()` method.

yes you're right @chrisvire ! i think because the types is using function overloading, we must declare the parameter type manually using `as` keyword

**Comment by oxwazz** [comment]
@ShayanAkhun this error shows up because your property is not found in any types in function overloading, you just change it the `auth.password` to `auth.pass` and the errors gone.

**Comment by oxwazz** [comment]
> I am getting the same issue if I set tls options

can you reproduce the issue?

**Comment by dhruvil-6** [comment]
You can definitely try out this.

It really worked for me!

import nodemailer from "nodemailer";

export const transporter = nodemailer.createTransport({
  host: `${process.env.SMTP_HOST}`,
  port: Number(process.env.SMTP_PORT),
  secure: false,
  auth: {
    user: process.env.SMTP_USERNAME,
    pass: process.env.SMTP_PASSWORD,
  },
});

**Comment by RaPriG** [comment]
My solution: 

To use SMTPTransport.

Ej:
import nodemailer from "nodemailer";
import **SMTPTransport** from "nodemailer/lib/smtp-transport";

 const transporter = nodemailer.createTransport({
            host: "smtp.example.com",
            port: 587,
            secure: false, // upgrade later with STARTTLS
            auth: {
                type: "OAuth2",
                user: config.senderEmail,
                clientId: config.gmailIdClient,
                clientSecret: config.gmailSecretClient,
                refreshToken: config.gmailRefreshToken,
                accessToken: accessToken || '',
            },
            tls: {
                rejectUnauthorized: false
            },
            logger: true
        } **as SMTPTransport.Options**);

**Comment by Myselfkc21** [comment]
> SO im doing a project for myself and when creating a transporter the host throws the following error **No overload matches this call. The last overload gave the following error. Argument of type '{ host: string; port: number; service: string | undefined; auth: { user: string | undefined; password: string | undefined; }; }' is not assignable to parameter of type 'TransportOptions | Transport'. Object literal may only specify known properties, and 'host' does not exist in type 'TransportOptions | Transport'.ts(2769) index.d.ts(77, 17): The last overload is declared here. (property) SMTPConnection.Options.host?: string | undefined** and my code is
> 
> ```
>  require("dotenv").config()
> import nodemailer, {Transporter} from 'nodemailer'
> import ejs from 'ejs'
> import path from 'path'
> 
> interface EmailOptions {
>     email:string,
>     subject:string,
>     template: string,
>     data :{[key:string]:any}
> }
> ```
> 
> ```
> 
> const sendMail = async (options: EmailOptions) : Promise <void> => {
>     const transporter : Transporter = nodemailer.createTransport({
>         host : process.env.SMTP_HOST as string,
>         port: parseInt(process.env.SMTP_PORT || '587') ,
>         service: process.env.SMTP_SERVICE ,
>         auth : {
>             user: process.env.SMTP_EMAIL,
>             password: process.env.SMTP_PASSWORD,
>         }
>     })
>     const {email ,subject , template , data} = options
> 
>     //get the path to the email template
>     const templatePath = path.join(__dirname, '../mails', template);
> 
>     //render the email template with ejs
>     const html:string = await ejs.renderFile(templatePath,data)
> 
>     const mailOptions = {
>         from : process.env.SMPT_MAIL,
>         to: email,
>         subject,
>         template
>     }
>     await transporter.sendMail(mailOptions)
> }
> 
> export default sendMail
> ```
> 
> ![code error](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/76954640/1d426922-cb98-4424-b568-a48a6f3ae66e)

facing the same issue rn, did u get any solution?

**Comment by Zain-ul-din** [comment]
try this

```ts
 const transporter = nodemailer.createTransport({
    ...
  } as SMTPTransport.Options);
```

**Comment by dougg0k** [comment]
Possibly related to https://github.com/nodemailer/nodemailer/issues/1727

**Comment by dipok-1** [comment]
just try this:
       auth : {
            user: process.env.SMTP_EMAIL,
            pass: process.env.SMTP_PASSWORD, instead of password use the already given config in the nodemailer library 'pass'for password
        }
you can check it here: https://nodemailer.com/

for me it worked..

**Comment by VitorPio7** [comment]
> I am getting the same issue. I looked at the tests in the `@types/nodemailer` source: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/nodemailer/nodemailer-tests.ts
> 
> And figured out that I needed to use `SMTPTransport.Options`. This is my code from a SvelteKit application:
> 
> ```
> import nodemailer from 'nodemailer';
> import type SMTPTransport from 'nodemailer/lib/smtp-transport';
> import {
>     SMTP_SERVER_ADDRESS,
>     SMTP_SERVER_PORT,
>     SMTP_SERVER_USERNAME,
>     SMTP_SERVER_PASSWORD
> } from '$env/static/private';
> 
> const smtpConfig: SMTPTransport.Options = {
>     host: SMTP_SERVER_ADDRESS,
>     port: Number(SMTP_SERVER_PORT),
>     auth: {
>         user: SMTP_SERVER_USERNAME,
>         pass: SMTP_SERVER_PASSWORD
>     }
> };
> const transporter = nodemailer.createTransport(smtpConfig);
> 
> export default transporter;
> ```

it's working for me now, thank you!!!
